documentation/projects/proposals/additional_search_views/20230719-implementation_plan_additional_search_views.md

```diff
@@ -111,35 +118,156 @@ for fuzzy matches.
+
+#### Add `strategy` and `collectionParams` to `search` store
+
+The `strategy` parameter will be used to determine the API request query
+parameters. If it is set to `"search"`, then the API query will be created using
+the current approach. If it is set to `"collection"`, the query will be
+constructed using the new method (`buildCollectionQuery`) to set `collection`
+and other relevant parameters using the new `collectionParams` object in the
+store. It will _not_ be setting the filter parameters such as license or
+category.
+
+### 4. Create collection pages
+
+We should add the following pages:
+
+- `/image/collection.vue`
+- `/audio/collection.vue`
+
+#### Validation of the collection query parameters
+
+This page will use the `collection` middleware to validate the collection query
+parameters:
+
+- if `collection` is set to `creator`, there should also exist a `source`
+  parameter
+- `source` parameter should be validated to be an existing source using the
+  provider store
+- `creator` and `tag` parameters should be decoded using `decodeURIComponent`.
+
+#### Fetching the media
 
-### 2. New API endpoints
+This page should also update the state (`searchType`, `collectionParams` and
+`strategy`) in the `search` store and handle fetching using `mediaStore`'s
+`fetchMedia` method in the `useFetch` hook.
 
-The new routes should use path parameters instead of query parameters for the
-`tag`, `creator` and `source` values. This will make the URLs more readable,
-easier to share, will be easier to cache or perform cache invalidation required
-by #1969. The path parameters should be URL encoded to preserve special
-characters and spaces.
+The media collections should be updated (not as part of this project) to move
+the load more methods to the page that fetches the media (i.e., `search.vue` or
+`collection.vue`), and the `mediaStore` should be updated to remove the load
+more methods.
 
-Instead of using query strings, we can describe the resource via the path:
-`/<media type>/source/<source>/creator/<creator>` is very clean, easy to read
-and understand, and very easy to manage the cache for because it is a static
-path. The source page can use the same route by leaving off the creator. This
-removes the need to manage specific query params as well and would allow us to
-add querying within these routes more easily in the future behind the regular q
-parameter if we wanted.
+#### SEO
 
-For the tag route, the singular `tag` rather than plural `tags` should be used
-for legibility since we are presenting a single tag.
+This page should also have relevant SEO properties, and should be indexed by the
+search engines.
 
-The new views should use the same pagination and dead link cleanup as the search
-views.
+The following titles should be used for the pages:
 
-### 3. New and updated components
+- "Images by Olga in Flickr" for the creator page
+- "Audio from Wikimedia" for the source page
+- "Images with the cat tag" for the tag page
```

Comment:
```suggestion
- "Cat Images" for the tag page
```

This might be more straightforward, what do you think?

Reply:
I was mainly thinking about the i18n difficulties when writing this sentence. For instance, to translate this sentence into Russian, you would have to change the case of "Cat" in this sentence. However, we could also add a note for translators to translate the "Images with the cat tag" sentence if translating "Cat images" would require to change the case of the tag itself (since changing the case of the tag name can't be automated, the "Cat images" would be wrong grammatically in the languages that use cases).
So, do you think changing this sentence to "Cat Images", and adding a comment to translators to translate this sentence as "Images with the Cat tag" if "Cat images" requires changes to the word "Cat", would work, @zackkrida ?

Reply:
Ah, that is indeed complicated. Perhaps for this plan all we need is a note that there _are_ i18n considerations in developing these strings, and we can figure out the specifics as we work on it.